addons/stock/models/product.py

```diff
@@ -191,6 +194,27 @@ def _compute_quantities_dict(self, lot_id, owner_id, package_id, from_date=False
 
         return res
 
+    def _compute_nbr_moves(self):
+        res = {i: {} for i in self.ids}
```

Comment:
With a defaultdict, it should be cleaner: `res = defaultdict(dict)` or `res = defaultdict(lambda: {'moves_in': 0, 'moves_out': 0})`. It is also true for product_template and avoid weird `if not template.id:`.
What do you think ?

Reply:
Agreed, I'm going to use both `res = defaultdict(dict)` (for product) and `res = defaultdict(lambda: {'moves_in': 0, 'moves_out': 0})` (for template)